### PR TITLE
Add 'ark restic init-repository' command

### PR DESCRIPTION
**Pull request by ncdc**

### CODE REVIEW

**Comment by ncdc** [comment]
This adds `ark restic init-repository`. We should probably have a command to rotate the key too, which will require some logic in the server (add new secret probably using a temp name, add that key to restic, update actual secret with new key, delete temp secret)

File: Makefile [hunk 1]
```diff
@@ -106,6 +106,7 @@ shell: build-dirs build-image
 		-v "$$(pwd)/.go/pkg:/go/pkg:delegated" \
 		-v "$$(pwd)/.go/std:/go/std:delegated" \
 		-v "$$(pwd):/go/src/$(PKG):delegated" \
+		-v "$$(pwd)/vendor/k8s.io/api:/go/src/k8s.io/api:delegated" \
```
Comment (skriss) [inline Makefile]: I added this already (https://github.com/skriss/ark/pull/1/files#diff-b67911656ef5d18c4ae36cb6741b7965R105) so can remove this one

File: pkg/cmd/cli/restic/init_repository.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	flags.StringVar(&o.KeyData, "key-data", o.KeyData, "Encryption key for the restic repository. Optional; if unset, Ark will generate a random key for you.")
+	flags.IntVar(&o.KeySize, "key-size", o.KeySize, "Size of the generated key for the restic repository")
+}
+
+func (o *InitRepositoryOptions) Complete(f client.Factory) error {
+	if o.KeyFile != "" && o.KeyData != "" {
+		return errors.Errorf("only one of --key-file and --key-data may be specified")
+	}
+
+	if o.KeySize < 1 {
```
Comment (skriss) [inline pkg/cmd/cli/restic/init_repository.go]: I think we only want to check this if not using `KeyFile` or `KeyData`, right?

**Review by skriss — CHANGES_REQUESTED** [review]
Couple comments, and docs/dev hardcoding need to be removed, but generally LGTM

File: pkg/cmd/cli/restic/init_repository.go (hunk 2 above)
Comment (ncdc) [reply pkg/cmd/cli/restic/init_repository.go]: Right

**Pull request closed without merging**